### PR TITLE
Add explicit error for PEP 585 type hints

### PR DESCRIPTION
**Pull request by jrmccluskey**
Adds an explicit warning for users attempting to use [PEP 585](https://peps.python.org/pep-0585/) type hints, directing them to use the typing module. Serves as a small improvement to the user experience while support for the new type hinting convention is designed and implemented.

Part of #23366

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
Run Python PreCommit

**Comment by jrmccluskey** [comment]
Run Python PreCommit

**Comment by jrmccluskey** [comment]
Run Python PreCommit

**Comment by jrmccluskey** [comment]
Run PythonDocs PreCommit

**Comment by jrmccluskey** [comment]
Run Python_Runners PreCommit

File: sdks/python/apache_beam/typehints/typehints.py [hunk 1]
```diff
@@ -1190,6 +1190,12 @@ def normalize(x, none_as_type=False):
     return type(None)
   elif x in _KNOWN_PRIMITIVE_TYPES:
     return _KNOWN_PRIMITIVE_TYPES[x]
+  elif sys.version_info >= (3, 9) and isinstance(x, types.GenericAlias):
+    # TODO(https://github.com/apache/beam/issues/23366): handle PEP 585
+    # generic type hints properly
+    raise TypeError(
+        'PEP 585 generic type hints like %s are not yet supported, '
+        'use typing module containers instead.' % x)
```
Comment (damccorm) [inline sdks/python/apache_beam/typehints/typehints.py:1198]: Could we drop a link to either docs on typing module containers?

**Review by damccorm — APPROVED** [review]
LGTM, just had one comment

File: sdks/python/apache_beam/typehints/typehints.py (hunk 1 above)
Comment (jrmccluskey) [reply sdks/python/apache_beam/typehints/typehints.py]: Done

**Review by damccorm — APPROVED** [review]
LGTM

**Comment by jrmccluskey** [comment]
Run Python_Runners PreCommit

**Pull request merged**